### PR TITLE
feat: Implement configurable minimum number of readers for popular tables

### PR DESCRIPTION
**Pull request by jacobhjkim**
### Summary of Changes

Related issue: [#534](https://github.com/lyft/amundsen/issues/534)

This PR adds `POPULAR_TABLE_MINIMUM_READER_COUNT` to the config file and  `num_readers` variable for the neo4j popular table query. This allows the minimum number of readers for popular tables to be configurable.

### Tests

N/A

### Documentation

No documentation changes at this time

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

### CODE REVIEW

**Comment by jornh** [comment]
Thanks for picking it up @jacobhjkim LGTM 👍  

Need Tao or someone else from Lyft to merge it.

**Comment by mgorsk1** [comment]
I am adding FrequentUsers functionality for AtlasProxy (PR 147) now and was thinking on something similar, will wait with my merge until this one is in :crossed_fingers:

**Comment by feng-tao** [comment]
sorry for the delay in review, will take a look!

File: metadata_service/config.py [hunk 1]
```diff
@@ -64,6 +64,9 @@ class Config:
     # On User detail method, these keys will be added into amundsen_common.models.user.User.other_key_values
     USER_OTHER_KEYS = {'mode_user_id'}  # type: Set[str]
 
+    # Number of minimum reader count to qualify for popular table
+    POPULAR_TABLE_MINIMUM_READER_COUNT = 4  # type: int
```
Comment (feng-tao) [inline metadata_service/config.py:68]: could we set the default to 10 which keeps the same as before? Users could modify it on its own config?

Comment (feng-tao) [reply metadata_service/config.py:68]: ^^ @jacobhjkim

File: metadata_service/config.py [hunk 2]
```diff
@@ -64,6 +64,9 @@ class Config:
     # On User detail method, these keys will be added into amundsen_common.models.user.User.other_key_values
     USER_OTHER_KEYS = {'mode_user_id'}  # type: Set[str]
 
+    # Number of minimum reader count to qualify for popular table
+    POPULAR_TABLE_MINIMUM_READER_COUNT = 10  # type: int
```
Comment (jacobhjkim) [inline metadata_service/config.py:68]:
```python
POPULAR_TABLE_MINIMUM_READER_COUNT = os.environ.get('POPULAR_TABLE_MINIMUM_READER_COUNT', 10)  # type: int
```
It should have been this. 🙃

Comment (feng-tao) [reply metadata_service/config.py:68]: not sure, but we could override it through different config right?

Comment (jacobhjkim) [reply metadata_service/config.py:68]: Right, production deployment should have some other config class. False alarm

**Pull request merged**